### PR TITLE
[PVT#180375608] Compile assets once on deploy and sync to/from S3

### PR DESCRIPTION
**Pull request by andyzito**
- Adds a new image variant (`--deploy`) for the deploy tasks (same as `--base` but with awscli installed)
- Deploy tasks now precompile assets and use `aws s3 sync` to push them to the `openpath-precompiled-assets` bucket at a path like `qa-warehouse-staging/84772df/` (`<targetgroup>/<githash>/`)
- The `entrypoint.sh` script now runs `assets:clobber` and then uses the existing `sync_app_assets.rb` script to pull down the precompiled assets from S3
- `config.assets.check_precompiled_assets` is now set to `false` (previously was defaulting to `true`). 
  - This was in response to an issue where Sprockets was unhappy with the synced assets and would recompile them on first page load (causing intermittent 504's for a while after)
  - Neither Elliot nor I were able to determine exactly what about the synced assets was making sprockets unhappy (that is, how does it know they aren't quite right), but turning off this precompiled check seems to force sprockets to just use the assets it has. 
  - I think it's pretty safe, and therefore not worth dumping more hours into figuring out why sprockets is unhappy with the synced assets and if there's another way to fix that -- but I'm open to dissenting opinions
  - github issue on sprockets-rails indicating similar behavior/resolution: https://github.com/rails/sprockets-rails/issues/352

This is deployed to QA as of this writing and appears to be functional.

### CODE REVIEW

File: config/environments/staging.rb
Comment (eanders) [inline config/environments/staging.rb:34]: Do we need this in `config/environments/production.rb` as well?

File: config/environments/development.rb
Comment (eanders) [inline config/environments/development.rb]: Is this still necessary?  Also, maybe not `config.assets.check_precompiled_asset = false` for development?

File: config/environments/staging.rb
Comment (andyzito) [reply config/environments/staging.rb:34]: yup

File: config/environments/development.rb
Comment (andyzito) [reply config/environments/development.rb]: No, and sure

**Comment by eanders** [comment]
@yakloinsteak, this looks good to me, and has been deployed to both QA and TC staging.  Could you give it a once over at some point just to see if I'm missing anything?

**Comment by andyzito** [comment]
Note: this is blocked by https://github.com/greenriver/ansible-fresh/pull/58

**Review by yakloinsteak — APPROVED** [review]
Looks good to me. I didn't test-drive it, but it all seems reasonable.

**Comment by andyzito** [comment]
@eanders The Ansible changes that were blocking this have been deployed to staging & production. Do you feel ready to merge and deploy this? Might recommend testing it on one non-QA client first before sending it everywhere, just in case.

**Comment by eanders** [comment]
Thanks @andyzito, I'll test and deploy later this week.  I think generally it's fine, but want to run a few more tests before merging.

**Pull request merged**
